src/components/tree/Tree.tsx

```diff
@@ -1,79 +1,141 @@
-import React, { FC } from "react";
+import React, { FC, useCallback, useState } from 'react';
+import sortBy from 'lodash.sortby';
 
-import CopyImage from "../vectors/CopyImage";
-import StarImgage from "../vectors/StarImage";
+import CopyImage from '../vectors/CopyImage';
+import StarImgage from '../vectors/StarImage';
+import StarOutline from '../vectors/StarOutline';
 
-import Reference from "./Reference";
-import { Article } from "../../utils/customTypes";
+import Reference from './Reference';
+import { Article } from '../../utils/customTypes';
 
-import "./Tree.css";
-import DATA from "./data.json";
+import './Tree.css';
+import DATA from './data.json';
 
 const INFO: {
   [key: string]: { title: string; info: string };
 } = {
   root: {
-    title: "Root",
+    title: 'Root',
     info: `
       Here you should find seminal articles from the original articles of
       your topic of interest.
     `,
   },
   trunk: {
-    title: "Trunk",
+    title: 'Trunk',
     info: `
       Here you should find articles where your topic of interest got a
       structure, these should be the first authors to discover the
       applicability of your topic of interest
     `,
   },
   leaf: {
-    title: "Leaves",
+    title: 'Leaves',
     info: `
       Here you should find recent articles and reviews that should
       condense very well your topics.
     `,
   },
 };
 
+const DEFAULT_SHOW = {
+  root: true,
+  trunk: true,
+  leaf: true,
+};
+
 const Tree: FC<{}> = () => {
   const data: { [section: string]: Article[] } = DATA;
+  const [star, setStar] = useState<{ [label: string]: boolean }>({});
+  const [show, setShow] = useState<{ [section: string]: boolean }>(
+    DEFAULT_SHOW
+  );
+
+  const toggleStar = useCallback((label: string) => {
+    setStar((current) => ({ ...current, [label]: !current[label] }));
+  }, []);
```

Comment:
There is an unexpected behavior when you click on a star, maybe because the position is changed because of this. Look at the video:
https://vimeo.com/user124877291/review/467130215/08f4b01cf5
A possible option would be this:
```suggestion
  const toggleStar = useCallback((label: string) => {
    setStar((current) =>
      Object.fromEntries(
        Object.entries(current).map((pair) => [
          pair[0],
          pair[0] === label ? !pair[1] : pair[1],
        ])
      )
    );
  }, []);
```